sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -888,6 +888,178 @@ case class MapFromEntries(child: Expression)
     copy(child = newChild)
 }
 
+@ExpressionDescription(
+  usage = """
+    _FUNC_(map[, ascendingOrder]) - Sorts the input map in ascending or descending order
+      according to the natural ordering of the map keys. The sorting algorithm used is
+      an adaptive, stable and iterative merge sort algorithm. If the input map is empty,
+      function returns an empty map.
+  """,
+  arguments =
+    """
+    Arguments:
+      * map - an expression. The map that will be sorted.
+      * ascendingOrder - an expression. The ordering in which the map will be sorted.
```

Comment:
All args are expression in general. Could you mention that it is a boolean parameter, and `true` means ascending order (apparently `false` is descending order).

Reply:
I was following the template for `Sequence` `ExpressionDefinision` which states `- an expression`.

Restructured the docs a bit now.